File: python/633. Sum of Square Numbers.py
class Solution(object):
    def judgeSquareSum(self, c):
        """
        :type c: int
        :rtype: bool
        """
        i = 0
        j = int(math.sqrt(c))
        while i<=j:
            total = i*i + j*j
            if total==c:
                return True
            elif total<c:
                i += 1
            else:
                j -= 1
        return False

File: go/src/leetcode/633. Sum of Square Numbers/633. Sum of Square Numbers.go
package main

import "math"

func judgeSquareSum(c int) bool {
	l := 0
	r := int(math.Sqrt(float64(c)))
	var d int
	for l <= r {
		d = l*l + r*r
		switch {
		case d == c:
			return true
		case d > c:
			r--
		case d < c:
			l++
		}
	}
	return false
}
